addons/account/models/account_invoice.py

```diff
@@ -1778,6 +1785,29 @@ def _prepare_invoice_line(self):
         }
         return data
 
+    @api.model
+    def create(self, vals):
+        if vals.get('display_type', self.default_get(['display_type'])['display_type']):
+            vals.update(price_unit=0, account_id=None, quantity=0)
+
+        return super(AccountInvoiceLine, self).create(vals)
+
+    @api.multi
+    def write(self, values):
+        if 'display_type' in values and self.filtered(lambda line: line.display_type != values.get('display_type')):
+            raise UserError("You cannot change the type of an invoice line.")
```

Comment:
Maybe something like "You cannot change the type of an invoice line. Create new sections or notes with new lines." ? In case you agree, to propagate to other check of display type in various models.

Reply:
Updated with " Instead you should delete the current line and create a new line of the proper type."